Windows registry value not found - PdhUtil.ENGLISH_COUNTER_KEY

Hello,

I have run into an issue on a Windows 7 machine while trying to get the processor information using the following code:
```
SystemInfo si = new SystemInfo();
si.getHardware().getProcessor();
```
Here are the relevant parts of the stack trace I got:

> java.lang.ExceptionInInitializerError
>     at oshi.hardware.platform.windows.WindowsHardwareAbstractionLayer.getProcessor(WindowsHardwareAbstractionLayer.java:64)
> ...
> Caused by: com.sun.jna.platform.win32.Win32Exception: The system cannot find the file specified.
>     at com.sun.jna.platform.win32.Advapi32Util.registryGetStringArray(Advapi32Util.java:800)
>     at com.sun.jna.platform.win32.Advapi32Util.registryGetStringArray(Advapi32Util.java:775)
>     at oshi.jna.platform.windows.PdhUtil.PdhLookupPerfIndexByEnglishName(PdhUtil.java:99)
>     at oshi.hardware.platform.windows.WindowsCentralProcessor.<clinit>(WindowsCentralProcessor.java:68)
>     ... 20 more


This machine does not have the registry value that Oshi tries to look for: _HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Perflib\009\Counter_.

The OS is in English, and the expected value exists in _HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Perflib\CurrentLanguage_.
**Ps.** All the machines I could get my hands on had the Counter value in this particular key (Windows 7 and Windows 10).

I was wondering if there is any particular reason to look for the key _009_. I believe that machines running in different languages would have the same problem.

Thank you!

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,7 @@
 3.9.0 (in progress)
 ================
 * [#630](https://github.com/oshi/oshi/pull/630): Add Sound Card list. - [@bilalAM](https://github.com/bilalAM).
+* [#636](https://github.com/oshi/oshi/pull/636): Catch exception when english counters missing. - [@dbwiddis](https://github.com/dbwiddis).
 * Your contribution here.
 
 3.8.1 (09/01/2018), 3.8.2 (09/07/2018), 3.8.3 (09/14/2018), 3.8.4 (09/24/2018)

--- a/oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsCentralProcessor.java
+++ b/oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsCentralProcessor.java
@@ -27,6 +27,7 @@ import com.sun.jna.Native; //NOSONAR
 import com.sun.jna.platform.win32.Advapi32Util;
 import com.sun.jna.platform.win32.Kernel32;
 import com.sun.jna.platform.win32.Kernel32Util;
+import com.sun.jna.platform.win32.Win32Exception;
 import com.sun.jna.platform.win32.WinBase;
 import com.sun.jna.platform.win32.WinBase.SYSTEM_INFO;
 import com.sun.jna.platform.win32.WinNT;
@@ -63,9 +64,23 @@ public class WindowsCentralProcessor extends AbstractCentralProcessor {
     // Save Windows version info for 32 bit/64 bit branch later
     private static final byte MAJOR_VERSION = Kernel32.INSTANCE.GetVersion().getLow().byteValue();
 
+    // Localize the "Processor" counter string. English counter names should
+    // normally be in HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows
+    // NT\CurrentVersion\Perflib\009\Counter, but language manipulations may
+    // delete the 009 index. In this case we can assume English must be the
+    // language and continue. We may still fail to match the name if the
+    // assumption is wrong but it's better than nothing.
     private static final String PROCESSOR = "Processor";
-    private static final String PROCESSOR_LOCALIZED = PdhUtil.PdhLookupPerfNameByIndex(null,
-            PdhUtil.PdhLookupPerfIndexByEnglishName(PROCESSOR));
+    private static final String PROCESSOR_LOCALIZED;
+    static {
+        String localized;
+        try {
+            localized = PdhUtil.PdhLookupPerfNameByIndex(null, PdhUtil.PdhLookupPerfIndexByEnglishName(PROCESSOR));
+        } catch (Win32Exception e) {
+            localized = PROCESSOR;
+        }
+        PROCESSOR_LOCALIZED = localized;
+    }
     private static final String TOTAL_INSTANCE = "_Total";
 
     /*

